experimental/bazel-base/Dockerfile

```diff
@@ -14,6 +14,8 @@ RUN apt-get update && \
         build-essential \
         ca-certificates \
         curl \
+        # dependency of @sentry_native//:sentry
+        libcurl4-openssl-dev \
```

Comment:
should we bazelfy transitive dependencies? (a quick googling shows that this might be a large task, so most likely not something to be done in this PR)

Reply:
yes we could. Theoretically, we should be able to just build curl with bazel, and point the paths out so that it's useable in sentry's cmake. probably, a lot of work though. 

I am trying to keep the dependencies in this dockerfile as small as possible so that we have an easier time bazelifying the leftover dependencies.